comp/core/autodiscovery/stream/stream.go

```diff
@@ -0,0 +1,75 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package stream provides the autodiscovery stream config
+package stream
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery"
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/proto"
+	pb "github.com/DataDog/datadog-agent/pkg/proto/pbgo/core"
+	"github.com/DataDog/datadog-agent/pkg/util/grpc"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// Config streams autodiscovery configs
+func Config(ac autodiscovery.Component, out pb.AgentSecure_AutodiscoveryStreamConfigServer) error {
+	s := &scheduler{
+		out:  out,
+		done: make(chan error, 1),
+	}
+
+	// TODO: add a uuid to avoid name collision when there are concurrent rpc calls ?
+	schedulerName := "remote"
+	ac.AddScheduler(schedulerName, s, true)
+	defer ac.RemoveScheduler(schedulerName)
+
+	return <-s.done
```

Comment:
Would this block?

Reply:
It blocks until the `scheduler` is done, which happens either when its `Close` method is called, or if there is an error (with the current implementation, might change)

Reply:
@pgimalac reminded me that each request is handle on a separate goroutine, so it does not prevent concurrent request from happening. 

Note to self: I sometime forget that Go handles each request on a separate goroutine